### ISSUE TITLE
Add Enumerable#without(*elements)

### ISSUE THREAD

**Issue by dhh** [rails/rails#19082]
Convenient short-form I've been using over and over again:

```ruby
module Enumerable
  def without(*elements)
    reject { |element| element.in?(elements) }
  end
end
```

May well be a more efficient way of doing this. I used to do stuff like `account.users - [ Current.user ]`.

**Comment by sgrif** [comment]
Is there a case where `ary.without(1, 2, 3)` is different than `ary - [1, 2, 3]`?

**Comment by dhh** [comment]
There’s no functional difference intended. This is all about aesthetics.

**Comment by todd** [comment]
I wouldn't mind taking this and exploring it a bit.

**Comment by dhh** [comment]
Have at it!

**Comment by JuanitoFatas** [comment]
This should be closed by #19157.
